lib/sales_engine.rb

```diff
@@ -134,4 +134,11 @@ def find_best_day_for_item(invoices)
     end
     top_invoice_item.created_at
   end
+
+  def favorite_customer(id)
+    cust_id = invoice_repository.find_invoices_by_merchant_id(id).group_by do |invoice|
+      invoice.customer_id
+    end.max_by { |k, v| v.size }.first
```

Comment:
Unused block argument - `k`. If it's necessary, use `_` or `_k` as an argument name to indicate that it won't be used.<br>Avoid multi-line chains of blocks.